api/src/pcapi/core/offers/repository.py

```diff
@@ -751,10 +749,17 @@ def get_filtered_stocks(
             sa.cast(models.Stock.beginningDatetime, sa.Time) >= time.replace(second=0),
             sa.cast(models.Stock.beginningDatetime, sa.Time) <= time.replace(second=59),
         )
-    query = _order_stocks_by(query, order_by, order_by_desc)
+    return _order_stocks_by(query, order_by, order_by_desc)
+
+
+def get_paginated_stocks(
+    stocks: flask_sqlalchemy.BaseQuery,
+    stocks_limit_per_page: int = LIMIT_STOCKS_PER_PAGE,
+    page: int = 1,
+) -> flask_sqlalchemy.BaseQuery:
     if FeatureToggle.WIP_PRO_STOCK_PAGINATION.is_active():
-        query = query.offset((page - 1) * stocks_limit_per_page).limit(stocks_limit_per_page)
-    return query.all()
+        return stocks.offset((page - 1) * stocks_limit_per_page).limit(stocks_limit_per_page)
+    return stocks
```

Comment:
`if not FeatureToggle.WIP_PRO_STOCK_PAGINATION.is_active():
    return stocks
return stocks.offset((page - 1) * stocks_limit_per_page).limit(stocks_limit_per_page)
`
Je dirais ça c'est mieux comme tu supprime tout le block quand le FF sera enlevé